modules/apps/portal-crypto-hash/portal-crypto-hash-api/src/main/java/com/liferay/portal/crypto/hash/verification/context/HashVerificationContext.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.crypto.hash.verification.context;
+
+import com.liferay.portal.crypto.hash.flavor.HashFlavor;
+
+import java.util.Optional;
+
+import org.json.JSONObject;
+
+/**
+ * @author Carlos Sierra Andrés
+ */
+public interface HashVerificationContext {
+
+	public HashFlavor getHashFlavor();
+
+	public Optional<JSONObject> getHashProviderMeta();
+
+	public String getHashProviderName();
+
+	public interface Builder extends PepperedBuilder {
+
+		public HashVerificationContext build(String serializedHashFlavor);
+
+	}
```

Comment:
I believe we should not have this if we don't have the equivalent in the generation, should we?

either we:
* pass the explicit pepper both in the generation and verification
* pass the pepper storage name so the library can handle it from the specified storage
* don't pass anything here in the builders and use whatever is registered in the library for the pepper